internal/overlord/backend.go

```diff
@@ -34,3 +34,15 @@ func (osb *overlordStateBackend) Checkpoint(data []byte) error {
 func (osb *overlordStateBackend) EnsureBefore(d time.Duration) {
 	osb.ensureBefore(d)
 }
+
+type nilStateBackend struct {
```

Comment:
`nil` has a special meaning in Go, and a struct isn't nil, so this name is a bit confusing. How about `noopStateBackend`?